server/block/break_info.go

```diff
@@ -236,6 +237,99 @@ func silkTouchOnlyDrop(it world.Item) func(t item.Tool, enchantments []item.Ench
 	}
 }
 
+// fortuneLevel returns the level of the fortune enchantment in enchantments, or 0 if it isn't present.
+func fortuneLevel(enchantments []item.Enchantment) int {
+	index := slices.IndexFunc(enchantments, func(i item.Enchantment) bool {
+		return i.Type() == enchantment.Fortune
+	})
+	if index == -1 {
+		return 0
+	}
+	return enchantments[index].Level()
+}
+
+// hasFortune checks if an item has the fortune enchantment.
+func hasFortune(enchantments []item.Enchantment) bool {
+	return slices.IndexFunc(enchantments, func(i item.Enchantment) bool {
+		return i.Type() == enchantment.Fortune
+	}) != -1
+}
+
+func fortuneOreCount(base int, enchantments []item.Enchantment) int {
+	if !hasFortune(enchantments) {
+		return base
+	}
+	fortune := fortuneLevel(enchantments)
+	if rand.IntN(fortune+2) < 2 {
+		return base
+	}
+	multiplier := rand.IntN(fortune) + 2
+	return base * multiplier
+}
+
+func fortuneDiscreteCount(minCount, maxCount, capCount int, enchantments []item.Enchantment) int {
+	fortune := 0
+	if hasFortune(enchantments) {
+		fortune = fortuneLevel(enchantments)
+	}
+	maxWithFortune := maxCount + fortune
+	if maxWithFortune > capCount {
+		maxWithFortune = capCount
+	}
+	return rand.IntN(maxWithFortune-minCount+1) + minCount
```

Comment:
From my understanding of wiki, this doesn't seem quite right?

<img width="766" height="359" alt="image" src="https://github.com/user-attachments/assets/77e45909-c078-49fc-bca5-5de79705f3ee" />


At Fortune 3, the chances are more heavily weighted for 4 drops but here they would be equal between 2-4.

I believe it should look something more like


```golang
maxWithFortune  := maxCount + fortune
return min(capCount, rand.IntN(maxWithFortune-minCount+1) + minCount))
```